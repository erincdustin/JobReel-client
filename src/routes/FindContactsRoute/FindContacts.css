.contacts-search-results {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr;
    grid-gap: 1em;
}

.results-container {
    position: relative;
    grid-column-start: 3;
    grid-column-end: 9;
    border: 1px solid black;
    border-radius: 18px;
    height: 400px;
    position: relative;
    background: #3C3230;
    color: #eeebd9;
    text-align: center;
  }
  
  #go-back {
    position: absolute;
    left: 0;
    top: 0;
    color: #eeebd9;
  }
  
  #go-back:hover {
    color: #fc766a;
  }
  
  .results > h2 {
    text-align: center;
  }
  
  .results {
    max-height: 100%;
    overflow-y: auto;
    overflow-block: inherit;
  }