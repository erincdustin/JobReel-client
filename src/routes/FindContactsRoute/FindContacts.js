import React, { Component } from 'react';
import JobReelContext from '../../context/JobReelContext'
import JobReelService from '../../services/jobreel-api-service'
import ProfessionalContact from '../../components/ProfessionalContact/ProfessionalContact'
import SideNav from '../../components/SideNav/SideNav'
import MediaQuery from 'react-responsive';
import findContactsRouteImage from '../../assests/findContactsRouteImage.svg'
import TopNav from '../../components/TopNav/TopNav'
import './FindContacts.css';
import { Link } from 'react-router-dom'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

export default class FindContactsRoute extends Component {

  static contextType = JobReelContext

  state = {
    search: null,
    savedContactEmails: [],
    noResults: false,
  }

  componentDidMount() {
    // if (this.context.professionalsSearch) {
    //   this.props.history.push(`/professionalsearch`)
    // }
    setTimeout(() => {
      JobReelService.getSavedContacts()
        .then(res => {
          const savedContactEmails = res.map(contact => contact.email)
          let savedContactEmailsObj = {}
          savedContactEmails.forEach(email => {
            savedContactEmailsObj[email] = email
          })
          this.setState({ savedContactEmails: savedContactEmailsObj })
            })
      const search = this.context.professionalsSearch
      JobReelService.getProfessionalEmails(search)
        .then(data => {
          if (data.data.emails.length === 0) {
            this.setState({noResults : true })
          } else {
          console.log(data.data.emails[0])
          this.context.setProfessionals(data.data.emails)
          this.context.setFindContactsMetaData(data.meta)
          }
        })
    }, 500)
  }

  renderProfessionalContacts() {
    const { professionals = [], professionalsSearch = {} } = this.context
    const professionalList = professionals.map((professional, i) => {
      if (professional.first_name) {
        return <ProfessionalContact professional={professional} key={i} search={professionalsSearch} savedContactEmails={this.state.savedContactEmails}/>
      }
      return null           
    })
    return (
      <div className='results'>
        {professionalList}
      </div>
    )
  }

  renderNoResultsMessage() {
    return (
        <h2>
            Sorry no results were found from that search.
      </h2>
    )
}

  render() {
    return (
      <div className='contacts-search-results'>
        <div className='findContactsRouteImage'>
          <img src={findContactsRouteImage} alt='find-contacts-route-background'/>
        </div>
        <div className='title'>
          <h2>Find Contacts</h2>
        </div>
        <MediaQuery minDeviceWidth={961}>
          <SideNav/>
        </MediaQuery>
        <MediaQuery maxDeviceWidth={960}>
          <TopNav/>
        </MediaQuery>
        <div className='results-container'>
          <Link to={`/professionalsearch`} alt="goBack">
            <FontAwesomeIcon id='go-back' icon='times-circle' size='2x'/>
          </Link>
          {this.renderProfessionalContacts()}
          {this.state.noResults && this.renderNoResultsMessage()}
        </div>
        
      </div>
    )
  }
}