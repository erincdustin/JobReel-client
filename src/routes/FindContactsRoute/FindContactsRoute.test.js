import React from 'react'
import ReactDOM from 'react-dom'
import {BrowserRouter} from 'react-router-dom'
import FindContactsRoute from './FindContacts'

describe('FindContactsRoute smoke test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><FindContactsRoute /></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})