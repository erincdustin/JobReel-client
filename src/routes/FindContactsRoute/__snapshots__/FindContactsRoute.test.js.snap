// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FindContactsRoute snapshot test renders the UI as expected 1`] = `
<div
  className="contacts-search-results"
>
  <div
    className="title"
  >
    <h2>
      Find Contacts
    </h2>
  </div>
  <MediaQueryContextConsumer
    minDeviceWidth={961}
  >
    <SideNav />
  </MediaQueryContextConsumer>
  <MediaQueryContextConsumer
    maxDeviceWidth={960}
  >
    <TopNav />
  </MediaQueryContextConsumer>
  <div
    className="results-container"
  >
    <Link
      alt="goBack"
      to="/professionalsearch"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="times-circle"
        id="go-back"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size="2x"
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </Link>
    <div
      className="results"
    />
  </div>
</div>
`;
